app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/roadbook/RoadBookFragmentTest.kt

```diff
@@ -1051,6 +1116,86 @@ class RoadBookFragmentTest {
             .substringBeforeLast(":")
     }
 
+    private fun updateTimestampOfRecord(index: Int) {
+        if (index < 0 || index >= DestinationRecords.RECORDS.size) {
+            throw IllegalArgumentException("Index out of bounds")
+        }
+        onView(withText(DestinationRecords.RECORDS[index].destID)).check(matches(isDisplayed()))
+        swipeRightTheRecordAt(index)
+        onView(withId(R.id.editTextTimestamp)).perform(click())
+        onView(withText(timePickerUpdateBLabel)).perform(click())
+        onView(withText(R.string.edit_dialog_update_b)).perform(click())
+
+    }
+
+    private fun getRbViewModel(): RoadBookViewModel? {
+        var rbViewModel : RoadBookViewModel? = null
+        testRule.scenario.onActivity {
+            val fragment = it.supportFragmentManager.fragments.first() as NavHostFragment
+            fragment.let {
+                val curr =
+                    it.childFragmentManager.primaryNavigationFragment as RoadBookFragment
+                rbViewModel = curr.getRBViewModelForTest()
+            }
+        }
+        return rbViewModel
+    }
+
+    private fun endShift(){
+        onView(withId(R.id.finish_shift)).perform(click())
+        onView(withText(R.string.end_shift_dialog_title))
+            .check(matches(isDisplayed()))
+        onView(withId(android.R.id.button1)).perform(click())
+    }
+
+    private fun getShiftFromDb(dbShiftRef : DatabaseReference) : List<Shift>{
+        val future = CompletableFuture<List<Shift>>()
+        dbShiftRef.get().addOnSuccessListener { snapshot ->
+            var records : List<DestinationRecord>  = emptyList()
+            var user : User = User()
+            var date : Date = Date()
+            val shiftDb = snapshot.children.mapNotNull { shifts ->
+                shifts.children.forEach {
```

Comment:
Did you try here using directly .getValue(Shift::class.java) ? Instead of desalinizing the attributes separately.
It should work and might increase a bit the coverage if the Datastore did not pass in the deserialization part of your ShiftDateKSerializer.